server/src/test/java/org/opensearch/indices/replication/checkpoint/PublishCheckpointActionTests.java

```diff
@@ -73,21 +73,16 @@ public void tearDown() throws Exception {
         super.tearDown();
     }
 
-    public void testPublishCheckpointActionOnPrimary() throws InterruptedException {
+    public void testPublishCheckpointActionOnPrimary() throws InterruptedException, IOException {
         final IndicesService indicesService = mock(IndicesService.class);
 
         final Index index = new Index("index", "uuid");
         final IndexService indexService = mock(IndexService.class);
         when(indicesService.indexServiceSafe(index)).thenReturn(indexService);
 
-        final int id = randomIntBetween(0, 4);
-        final IndexShard indexShard = mock(IndexShard.class);
-        when(indexService.getShard(id)).thenReturn(indexShard);
+        final IndexShard indexShard = newShard(true);
```

Comment:
Why is it necessary for the `IndexShard` object to no longer be a mock in these tests?

Reply:
Now PublishCheckpointActionTests is extending IndexShardTestCase, so we can easily create a new sample IndexShard. The reason we were mocking IndexShard before was to just get some responses on certain method calls, this was solved by " newShard(true); ". So mock is no longer needed.

Reply:
Got it makes sense. IndexShard can be a mock, there is no need for a `newShard` mocked IndexShard instance would suffice for this unit tests. Thanks @kartg